system/lib/wasmfs/file.cpp

```diff
@@ -223,6 +235,20 @@ std::vector<Directory::Entry> Directory::Handle::getEntries() {
       entries.push_back({name, entry.file->kind, entry.file->getIno()});
     }
   }
+
+#ifdef WASMFS_CASE_INSENSITIVE
+  // Restore the original name of the entries (case preservation).
```

Comment:
Which option do you think is the best: normalizing in `Directory::Handle` or in a virtual backend?
Personally, I find virtual backend implementation more problematic since it's hard to maintain both backend directory entries and underlying directory entries. But maybe, it's only because it inherits from `MemoryDirectory`.
The version which uses `Directory::Handle` got deleted from this PR. But it's available here https://github.com/emscripten-core/emscripten/commit/1714fd76a6dd9a51d8340c9565e0af4cf547ec1e